quarkus/config-api/src/main/java/org/keycloak/config/DatabaseOptions.java

```diff
@@ -82,4 +87,119 @@ public class DatabaseOptions {
             .defaultValue(100)
             .description("The maximum size of the connection pool.")
             .build();
+
+    public static final Option<Boolean> DB_ENABLED_DATASOURCE = new OptionBuilder<>("db-enabled-<datasource>", Boolean.class)
+            .category(OptionCategory.DATABASE_DATASOURCES)
+            .defaultValue(true)
+            .description("Enable/disable specific named datasource <datasource>.")
+            .build();
+
+    /**
+     * Options that have their sibling for a named datasource
+     * Example: for `db-dialect`, `db-dialect-<datasource>` is created
+     */
+    public static final List<Option<?>> OPTIONS_DATASOURCES = List.of(
+            DatabaseOptions.DB_DIALECT,
+            DatabaseOptions.DB_DRIVER,
+            DatabaseOptions.DB,
+            DatabaseOptions.DB_URL,
+            DatabaseOptions.DB_URL_HOST,
+            DatabaseOptions.DB_URL_DATABASE,
+            DatabaseOptions.DB_URL_PORT,
+            DatabaseOptions.DB_URL_PROPERTIES,
+            DatabaseOptions.DB_USERNAME,
+            DatabaseOptions.DB_PASSWORD,
+            DatabaseOptions.DB_SCHEMA,
+            DatabaseOptions.DB_POOL_INITIAL_SIZE,
+            DatabaseOptions.DB_POOL_MIN_SIZE,
+            DatabaseOptions.DB_POOL_MAX_SIZE
+    );
+
+    /**
+     * In order to avoid ambiguity, we need to have unique option names for wildcard options.
+     * This map controls overriding option name to be unique for wildcard option.
+     */
+    private static final Map<String, String> DATASOURCES_OVERRIDES_SUFFIX = Map.of(
+            DatabaseOptions.DB.getKey(), "-kind", // db-kind
+            DatabaseOptions.DB_URL.getKey(), "-full"  // db-url-full
+    );
+
+    private static final Map<String, Option<?>> cachedDatasourceOptions = new HashMap<>();
+
+
+    /**
+     * Get datasource option containing named datasource mapped to parent DB options.
+     * <p>
+     * We map DB options to named datasource options like:
+     * <ul>
+     *     <li>{@code db-url-host --> db-url-host-<datasource>}</li>
+     *     <li>{@code db-username --> db-username-<datasource>}</li>
+     *     <li>{@code db --> db-kind-<datasource>}</li>
+     * </ul>
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> Optional<Option<T>> getDatasourceOption(Option<T> parentOption) {
+        if (!OPTIONS_DATASOURCES.contains(parentOption)) {
+            return Optional.empty();
+        }
+
+        var key = getKeyForDatasource(parentOption);
+        if (key.isEmpty()) {
+            return Optional.empty();
+        }
+
+        // check if we already created the same option and return it from the cache
+        Option<?> option = cachedDatasourceOptions.get(key.get());
+
+        if (option == null) {
+            var builder = new OptionBuilder<>(key.get(), parentOption.getType())
```

Comment:
We have something related with Option.withRuntimeSpecificDefault

Maybe in a follow up I'd like to consider having a general Option.toBuilder method instead.

Reply:
@shawkins Yes, that'd be better to not explicitly map all these attributes here. However, as you mentioned, we can do it in a follow-up task. Does it work for you?

Reply:
Sounds good

Reply:
It was quite low-hanging fruit, so added one other commit here: https://github.com/keycloak/keycloak/pull/37640/commits/4c251160ca8744cd61c6fae73c8ac5377d061858

Reply:
Perfect, thanks @mabartos